torch/_functorch/_aot_autograd/schemas.py

```diff
@@ -194,7 +194,45 @@ class SubclassCreationMeta:
     # Used at runtime to determine the subclass type, so we don't need to save the original subclass
     original_subclass_type: Optional[type] = None
 
-    def creation_fn(self, all_args, *, is_runtime: bool):
+    def compute_outer_size_and_stride(
+        self,
+        all_args,
+        *,
+        curr_start_idx: int,
+        is_runtime: bool,
+        is_subclass: bool,
+    ):
+        outer_size_symint_placeholders = [type(i) is not int for i in self.outer_size]
+        outer_stride_symint_placeholders = [
+            type(i) is not int for i in self.outer_stride
+        ]
+        has_symbolic = any(outer_size_symint_placeholders)
+
+        if is_runtime and has_symbolic and not is_subclass:
+            start = curr_start_idx
+            end = start + sum(outer_size_symint_placeholders)
+            it = iter(all_args[start:end])
+            outer_size = pytree.tree_map_only(
+                torch.SymInt, lambda _: next(it), self.outer_size
+            )
+
+            start = end
+            end = start + sum(outer_stride_symint_placeholders)
+            it = iter(all_args[start:end])
+            outer_stride = pytree.tree_map_only(
+                torch.SymInt, lambda _: next(it), self.outer_stride
+            )
+            return outer_size, outer_stride
+
+        return self.outer_size, self.outer_stride
+
+    def creation_fn(
+        self,
+        all_args,
+        *,
+        is_runtime: bool,
+        is_subclass: bool,
```

Comment:
I'm not sure what we need this extra `is_subclass` arg for, but it looks like it's not used anywhere?

Reply:
`is_subclass` was a poor choice of word. This is so that we can properly handle nested tensors.